### PR TITLE
Moved and revised the is operator article

### PR DESCRIPTION
**Pull request by pkulikov**
Removed the pattern matching details and added the links to the patterns article.
And, at last (!), moved the article to the operators section of the language reference.

### CODE REVIEW

File: docs/csharp/language-reference/operators/is.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Beginning with C# 7.0, you can also use the `is` operator to match an expression against a pattern, as the following example shows:
+
+:::code language="csharp" source="snippets/shared/IsOperator.cs" id="IntroExample":::
+
+In the preceding example, the `is` operator matches an expression against a [property pattern](patterns.md#property-pattern) with nested [constant](patterns.md#constant-pattern) and [relational](patterns.md#relational-patterns) patterns.
+
+The `is` operator can be useful in the following scenarios:
+
+- To check the runtime type of an expression, as the following example shows:
```
Comment (gewarren) [inline docs/csharp/language-reference/operators/is.md:24]:
```suggestion
- To check the run-time type of an expression, as the following example shows:
```

File: docs/csharp/language-reference/operators/type-testing-and-cast.md [hunk 2]
```diff
@@ -42,7 +42,17 @@ E is T
 
 where `E` is an expression that returns a value and `T` is the name of a type or a type parameter. `E` cannot be an anonymous method or a lambda expression.
 
-The `E is T` expression returns `true` if the result of `E` is non-null and can be converted to type `T` by a reference conversion, a boxing conversion, or an unboxing conversion; otherwise, it returns `false`. The `is` operator doesn't consider user-defined conversions.
+The `is` operator returns `true` when an expression result is non-null and any of the following conditions are true:
+
+- The runtime type of an expression result is `T`.
```
Comment (gewarren) [inline docs/csharp/language-reference/operators/type-testing-and-cast.md:47]:
```suggestion
- The run-time type of an expression result is `T`.
```

File: docs/csharp/language-reference/operators/type-testing-and-cast.md [hunk 3]
```diff
@@ -42,7 +42,17 @@ E is T
...
+
+- The runtime type of an expression result derives from type `T` or implements interface `T` or another [implicit reference conversion](~/_csharplang/spec/conversions.md#implicit-reference-conversions) exists from it to `T`.
```
Comment (gewarren) [inline docs/csharp/language-reference/operators/type-testing-and-cast.md:49]:
```suggestion
- The run-time type of an expression result derives from type `T`, implements interface `T`, or another [implicit reference conversion](~/_csharplang/spec/conversions.md#implicit-reference-conversions) exists from it to `T`.
```

File: docs/csharp/language-reference/operators/type-testing-and-cast.md [hunk 4]
```diff
@@ -42,7 +42,17 @@ E is T
...
+
+- The runtime type of an expression result is a [nullable value type](../builtin-types/nullable-value-types.md) with the underlying type `T`.
```
Comment (gewarren) [inline docs/csharp/language-reference/operators/type-testing-and-cast.md:51]:
```suggestion
- The run-time type of an expression result is a [nullable value type](../builtin-types/nullable-value-types.md) with the underlying type `T`.
```

File: docs/csharp/language-reference/operators/type-testing-and-cast.md [hunk 5]
```diff
@@ -42,7 +42,17 @@ E is T
...
+
+- A [boxing](../../programming-guide/types/boxing-and-unboxing.md#boxing) or [unboxing](../../programming-guide/types/boxing-and-unboxing.md#unboxing) conversion exists from the runtime type of an expression result to type `T`.
```
Comment (gewarren) [inline docs/csharp/language-reference/operators/type-testing-and-cast.md:53]:
```suggestion
- A [boxing](../../programming-guide/types/boxing-and-unboxing.md#boxing) or [unboxing](../../programming-guide/types/boxing-and-unboxing.md#unboxing) conversion exists from the run-time type of an expression result to type `T`.
```

File: docs/csharp/language-reference/operators/type-testing-and-cast.md [hunk 6]
```diff
@@ -42,7 +42,17 @@ E is T
...
+
+The `is` operator doesn't consider user-defined conversions.
 
 The following example demonstrates that the `is` operator returns `true` if the runtime type of an expression result derives from a given type, that is, there exists a reference conversion between types:
```
Comment (gewarren) [inline docs/csharp/language-reference/operators/type-testing-and-cast.md:57]:
```suggestion
The following example demonstrates that the `is` operator returns `true` if the run-time type of an expression result derives from a given type, that is, there exists a reference conversion between types:
```

File: docs/csharp/language-reference/operators/type-testing-and-cast.md [hunk 7]
```diff
@@ -42,9 +42,19 @@ E is T
...
 
-The `E is T` expression returns `true` if the result of `E` is non-null and can be converted to type `T` by a reference conversion, a boxing conversion, or an unboxing conversion; otherwise, it returns `false`. The `is` operator doesn't consider user-defined conversions.
+The `is` operator returns `true` when an expression result is non-null and any of the following conditions are true:
 
-The following example demonstrates that the `is` operator returns `true` if the runtime type of an expression result derives from a given type, that is, there exists a reference conversion between types:
+- The run-time type of an expression result is `T`.
+
+- The run-time type of an expression result derives from type `T`, implements interface `T`, or another [implicit reference conversion](~/_csharplang/spec/conversions.md#implicit-reference-conversions) exists from it to `T`.
+
+- The run-time type of an expression result is a [nullable value type](../builtin-types/nullable-value-types.md) with the underlying type `T`.
```
Comment (BillWagner) [inline docs/csharp/language-reference/operators/type-testing-and-cast.md:51]: The `HasValue` property must be true as well (which I know is covered in the non-null part, but it's worth stating:

```suggestion
- The run-time type of an expression result is a [nullable value type](../builtin-types/nullable-value-types.md) with the underlying type `T` and the <xref:System.Nullable%601.HasValue?displayProperty=nameWithType> is 1true`.
```

**Review by BillWagner — COMMENTED** [review]
Thanks @pkulikov 

I've got one suggestion to be explicit on the nullable value type conversion, and then this is ready to :shipit:

**Comment by BillWagner** [comment]
@pkulikov 

One other note, as you work on other PRs that move content and have redirections.  @gewarren has created a tool, [cleanrepo](https://github.com/gewarren/cleanrepo) that has two options that help a lot to find and fix any links to content that moves.  I've started using the `--replace-redirects` followed by the `--relative-links` options to update all links to redirected content.

**Comment by pkulikov** [comment]
@BillWagner thank you for the link to the tool!
I've applied your suggestion

**Review by BillWagner — APPROVED** [review]
Thanks again @pkulikov 

I'll :shipit: now.

**Pull request merged**